### PR TITLE
Migrate CMSRucioClient base image to almalinux9

### PR DESCRIPTION
**Pull request by d-ylee**
For #652

Migration of CMSRucioClient base image to Alma9. Build script for CMSRucioClient image.

### CODE REVIEW

**Comment by d-ylee** [comment]
Since rucio/containers is using dnf, should we also switch from yum to dnf?

File: docker/CMSRucioClient/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,6 @@
-FROM centos:7
+FROM almalinux:9
+
+ARG RUCIO_VERSION
 
 RUN yum upgrade -y && \
```
Comment (ericvaandering) [inline docker/CMSRucioClient/Dockerfile:5]: To be really compliant yum is replaced with dnf in alma9. As I understand it they are different programs but generally the interface is the same so s/yum/dnf/g should work.

**Comment by d-ylee** [comment]
@ericvaandering Should I also be updating Dockerfile.trace and Dockerfile.loadtest?

**Comment by ericvaandering** [comment]
> Since rucio/containers is using dnf, should we also switch from yum to dnf?

For sure. No, trace and load test are different. Loadtest is assigned to @haozturk and I'm not sure we use the trace docker file for anything now. The actual trace recipients are based on Alpine linux and Go

**Comment by d-ylee** [comment]
How do we go about testing the image?

**Comment by d-ylee** [comment]
I ran the pod and added my certs. I was able to get an X509 proxy and `rucio ping` and `rucio whoami` the `rucio-int` server.

File: docker/CMSRucioClient/Dockerfile [hunk 2]
```diff
@@ -69,14 +72,14 @@ COPY rucio-prod.cfg /opt/rucio-prod/etc/rucio.cfg
 WORKDIR /root
 RUN git clone https://github.com/dmwm/CMSRucio.git
 
-RUN mkdir -p /etc/grid-security ; ln -s /cvmfs/grid.cern.ch/etc/grid-security/certificates /etc/grid-security/
+# RUN mkdir -p /etc/grid-security ; ln -s /cvmfs/grid.cern.ch/etc/grid-security/certificates /etc/grid-security/
```
Comment (ericvaandering) [inline docker/CMSRucioClient/Dockerfile:75]: This will work in the production system, so let's put it back.

**Review by ericvaandering — CHANGES_REQUESTED** [review]

File: docker/CMSRucioClient/Dockerfile (hunk 2 above)
Comment (d-ylee) [reply docker/CMSRucioClient/Dockerfile]: Ok I reverted the change.

**Comment by d-ylee** [comment]
@ericvaandering 
I was looking at the github action to build and push images to the CERN registry.

I see that the action uses a matrix of the directory names in the `docker` directory to name the images:
https://github.com/dmwm/CMSRucio/blob/9b2e4df0f1129080a9b0de5242e5a0ddbf88f231/.github/workflows/rucio-release-images.yml#L14

https://github.com/dmwm/CMSRucio/blob/9b2e4df0f1129080a9b0de5242e5a0ddbf88f231/.github/workflows/rucio-release-images.yml#L38

In Harbor, I think the name of the image is `rucio_client`. Would we need to change the directory name from `CMSRucioClient` to `rucio_client` if we want to use the Github Action to automate building and pushing?

**Comment by ericvaandering** [comment]
> @ericvaandering I was looking at the github action to build and push images to the CERN registry.
> 
> I see that the action uses a matrix of the directory names in the `docker` directory to name the images:
> 
> https://github.com/dmwm/CMSRucio/blob/9b2e4df0f1129080a9b0de5242e5a0ddbf88f231/.github/workflows/rucio-release-images.yml#L14
> 
> https://github.com/dmwm/CMSRucio/blob/9b2e4df0f1129080a9b0de5242e5a0ddbf88f231/.github/workflows/rucio-release-images.yml#L38
> 
> In Harbor, I think the name of the image is `rucio_client`. Would we need to change the directory name from `CMSRucioClient` to `rucio_client` if we want to use the Github Action to automate building and pushing?

Sorry to get to this so late. Quit possibly. I think we have a separate issue for the automated builds. So refactoring it that way is fine.

**Review by ericvaandering — APPROVED** [review]

**Pull request merged**